src/components/Root/index.tsx

```diff
@@ -37,6 +43,11 @@ const RootConsumer = (): React.ReactElement | null => {
       } catch (err) {
         logError(Errors._904, err.message)
         setIsError(true)
+      } finally {
+        // If chain is not returned from CGW, revert to default
+        if (!isValidChainId(chainId)) {
```

Comment:
This useEffect will have to depend on the chainId then. Which is no bueno.

Reply:
Also, the logic that sets the initial chain is independent from loading chains, so it can cause a race condition.